crates/matrix-sdk-ui/tests/integration/timeline/pagination.rs

```diff
@@ -842,3 +849,246 @@ async fn test_back_pagination_aborted() {
     // And there should be no other pending pagination status updates.
     assert!(back_pagination_status.next().now_or_never().is_none());
 }
+
+#[async_test]
+async fn test_lazy_back_pagination() {
+    let room_id = room_id!("!foo:bar.baz");
+    let event_factory = EventFactory::new().room(room_id).sender(&ALICE);
+
+    let mock_server = MatrixMockServer::new().await;
+    let client = mock_server.client_builder().build().await;
+
+    let room = mock_server.sync_joined_room(&client, room_id).await;
+    let timeline = room.timeline().await.unwrap();
+    let (timeline_initial_items, mut timeline_stream) = timeline.subscribe().await;
+
+    assert!(timeline_initial_items.is_empty());
+    assert_pending!(timeline_stream);
+
+    // Receive 30 events.
+    mock_server
+        .mock_sync()
+        .ok_and_run(&client, |sync_builder| {
+            sync_builder.add_joined_room({
+                let mut room = JoinedRoomBuilder::new(room_id);
+
+                for nth in 0..30 {
+                    room = room.add_timeline_event(
+                        event_factory
+                            .text_msg("foo")
+                            .event_id(&EventId::parse(format!("$ev{nth}")).unwrap()),
+                    );
+                }
+
+                room.set_timeline_prev_batch("back_pagination_token_1").set_timeline_limited()
+            });
+        })
+        .await;
+
+    // Only 20 items are broadcasted to the subscriber.
+    assert_timeline_stream! {
+        [timeline_stream]
+        // Huh?
+        // TODO: understand why
+        clear;
+        clear;
+
+        // `$ev11`, the 1st item, and its update.
+        append "$ev11";
+        update[0] "$ev11";
```

Comment:
Why do we have an update? (I suspect this might be the read receipt being updated, but worth double-checking?)

Reply:
It's the read receipt, yup.